packages/wp-story-editor/src/components/postReviewDialog/index.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useEffect, useState } from '@googleforcreators/react';
+import { __ } from '@googleforcreators/i18n';
+import { Text, THEME_CONSTANTS } from '@googleforcreators/design-system';
+import { Dialog, useStory } from '@googleforcreators/story-editor';
+
+function PostReviewDialog() {
+  const { isFreshlyPending } = useStory(
+    ({
+      state: {
+        capabilities: { hasPublishAction },
+        meta: { isFreshlyPending },
+      },
+    }) => ({
+      hasPublishAction,
+      isFreshlyPending,
+    })
+  );
```

Comment:
You're right, I'll remove it!

Reply:
Oh yeah, my bad. Mostly just meant we can remove `capabilities: { hasPublishAction },`

Reply:
@littlemilkstudio Can you remind me why using the selector pattern is the way we prefer to go?

Reply:
yea absolutely. Basically `useContextSelector` only re-renders the component when the return value of the selector is shallowly equal between renders. so if you do something like this:
```js
const { state: { someStoryProperty } } = useStory();
```
this will re-render every time the story provider re-renders. 

However, if you do this:
```js
const { someStoryProperty } = useStory(({ state }) => ({ someStoryProperty: state.someStoryProperty }));
```
this will now only update when `someStoryProperty` updates between re-renders.

Here's a more in depth [synopsis](https://github.com/GoogleForCreators/web-stories-wp/issues/2662#issuecomment-1011372651)

TLDR; it functions the same, but is more performant to use a selector

Reply:
Ah, that is right! I knew it had something to do with re-rendering all the time verses only when `someStoryProperty` updates, similar to selectors in Redux. It had just been awhile since we talked about this, and I had forgotten the details. Thanks @littlemilkstudio you're the best.

Reply:
Super helpful context, thank you for this! And for Brooke asking the question too!